app/Http/Controllers/UserProductController.php

```diff
@@ -32,8 +34,20 @@ public function showDetails($id)
         $product = Product::findorfail($id);
         $brands = Brand::all();
         $images = Image::where('product_id', $product->id)->get();
+        $allowComment = false;
 
-        return view('users.show')->with(compact('product', 'images', 'brands'));
+        $user = User::with(['orders' => function ($query) {
+            $query->where('order_status_id', config('orderstatus.delivered'));
+        }])->where('id', Auth::user()->id)->first();
+        
+        foreach ($user->orders as $order) {
+            foreach ($order->products as $p) {
+                if ($p->id == $id) {
+                    $allowComment = "true";
```

Comment:
```suggestion
                    $allowComment = true;
```
sao lại text true nhỉ?